src/plugins/autocompletion_providers/Git.ts

```diff
@@ -34,11 +35,161 @@ function canonizeBranchAlias(alias: string) {
+        noShortFlag: true,
+    },
+    {
+        longFlag: "short",
+        description: "When doing a dry-run, give the output in the short-format. See git-status(1) for details. Implies --dry-run.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "branch",
+        description: "Show the branch and tracking info even in short-format.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "porcelain",
+        description: "When doing a dry-run, give the output in a porcelain-ready format. See git-status(1) for details. Implies --dry-run.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "long",
+        description: "When doing a dry-run, give the output in a the long-format. Implies --dry-run.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "allow-empty",
+        description: "Usually recording a commit that has the exact same tree as its sole parent commit is a mistake,\
+         and the command prevents you from making such a commit. This option bypasses the safety, and is primarily for\
+         use by foreign SCM interface scripts.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "allow-empty-message",
+        description: "Like --allow-empty this command is primarily for use by foreign SCM interface scripts. It allows\
+         you to create a commit with an empty commit message without using plumbing commands like git-commit-tree(1).",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "no-edit",
+        description: "Use the selected commit message without launching an editor. For example, git commit --amend --no-edit amends a commit without changing its commit message.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "no-post-rewrite",
+        description: "Bypass the post-rewrite hook.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "dry-run",
+        description: "Do not create a commit, but show a list of paths that are to be committed, paths with local changes that will be left uncommitted and paths that are untracked.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "status",
+        description: "Include the output of git-status(1) in the commit message template when using an editor to\
+         prepare the commit message. Defaults to on, but can be used to override configuration variable commit.status.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "no-status",
+        description: "Do not include the output of git-status(1) in the commit message template when using an editor to prepare the default commit message.",
+        noShortFlag: true,
+    },
+    {
+        longFlag: "no-gpg-sign",
+        description: "Countermand commit.gpgSign configuration variable that is set to force each and every commit to be signed.",
+        noShortFlag: true,
```

Comment:
That has a separate meaning

key present -> use it
key not present -> use the first letter of the long flag
noShortFlag: true -> there is no shortFlag